lms/djangoapps/django_comment_client/forum/views.py

```diff
@@ -258,15 +258,21 @@ def single_thread(request, course_key, discussion_id, thread_id):
     """
     Renders a response to display a single discussion thread.
     """
-
     nr_transaction = newrelic.agent.current_transaction()
 
     course = get_course_with_access(request.user, 'load_forum', course_key)
-    course_settings = make_course_settings(course)
+    course_settings = make_course_settings(course, request.user)
     cc_user = cc.User.from_django_user(request.user)
     user_info = cc_user.to_dict()
     is_moderator = cached_has_permission(request.user, "see_all_cohorts", course_key)
 
+    # Verify that student has access to this thread if belongs to a discussion module
+    if discussion_id not in course.top_level_discussion_topic_ids and \
```

Comment:
This logic was recommended to me by @jimabramson, who told me that any topic that is not in `course.top_level_discussion_topic_ids` is necessarily a discussion module.